web/yo/app/scripts/controllers/gene.js

```diff
@@ -21,8 +21,26 @@ angular.module('oncokbApp')
                     }
                 });
             };
-
+            function setUUID(obj) {
+                _.each(_.keys(_.clone(obj)), function(key) {
+                    if (key !== 'short') {
+                        obj[key + '_uuid'] = UUIDjs.create(4).toString();
+                        if (_.isObject(obj[key])) {
+                            setUUID(obj[key]);
+                        }
+                    }
+                });
+            }
             $scope.addMutation = function(newMutationName) {
+                // var mutation = _.clone(OncoKB.firebaseModel.Mutation);
+                // mutation.name = newMutationName;
+                // mutation.mutation_effect = _.clone(OncoKB.firebaseModel.MutationEffect);
+                // setUUID(mutation);
+                var mutation = new FirebaseModel.Mutation(newMutationName);
+                $scope.geneFire.mutations.push(mutation);
+                console.log('new mutation added');
```

Comment:
When we towards the end of development. We need to come back clean up all unnecessary console logs and commented out code.